patches/api/0308-ItemStack-editMeta.patch

```diff
@@ -27,9 +27,28 @@ index 86bd9f14de5c1ff3d797955be1af56e5efcac884..59a026d80b0a0a4890becf98efdbe532
 +     * @return {@code true} if the edit was successful, {@code false} otherwise
 +     */
 +    public boolean editMeta(final @NotNull java.util.function.Consumer<? super ItemMeta> consumer) {
++        return editMeta(ItemMeta.class, consumer);
++    }
++
++    /**
++     * Edits the {@link ItemMeta} of this stack if the meta is of the specified type.
++     * <p>
++     * The {@link java.util.function.Consumer} must only interact
++     * with this stack's {@link ItemMeta} through the provided {@link ItemMeta} instance.
++     * Calling this method or any other meta-related method of the {@link ItemStack} class
++     * (such as {@link #getItemMeta()}, {@link #addItemFlags(ItemFlag...)}, {@link #lore()}, etc.)
++     * from inside the consumer is disallowed and will produce undefined results or exceptions.
++     * </p>
++     *
++     * @param metaClass the type of meta to edit
++     * @param consumer the meta consumer
++     * @param <M> the meta type
++     * @return {@code true} if the edit was successful, {@code false} otherwise
++     */
++    public <M extends ItemMeta> boolean editMeta(final @NotNull Class<M> metaClass, final @NotNull java.util.function.Consumer<? super M> consumer) {
 +        final ItemMeta meta = this.getItemMeta();
-+        if (meta != null) {
-+            consumer.accept(meta);
++        if (meta != null && metaClass.isInstance(meta)) {
```

Comment:
Could just as well then remove the `meta != null` check, as this just doubles logic at this point.